constellations/gemini-admin/src/user/userAFH.js

```diff
@@ -0,0 +1,187 @@
+"use strict";
+require("./userAFH.sass");
+import React from "react";
+import moment from "moment";
+import axios from "axios";
+import { Link } from "react-router-dom";
+import API from "../api.js";
+
+export class UserAFHPage extends React.Component {
+    state = {
+        id: 0,
+        user: {},
+        userAFHs: [],
+        afhs: [],
+        afhId: 0,
+    };
+
+    componentDidMount = () => {
+        this.fetchUser();
+    };
+
+    fetchUser = () => {
+        const id = this.props.id;
+
+        const apiCalls = [
+            API.get("api/users/user/" + id),
+            API.get("api/userafhs/users/" + id),
+        ];
+
+        axios
+            .all(apiCalls)
+            .then(
+                axios.spread((...responses) => {
+                    let afhIds = [];
+                    responses[1].data.forEach((userAFH) => {
+                        afhIds.push(userAFH.afhId);
+                    });
+
+                    this.setState({
+                        id: id,
+                        user: responses[0].data,
+                    });
+                    this.fetchAFHs(afhIds);
+                })
+            )
+            .catch((err) => alert("Could not fetch user: " + err));
+    };
+
+    fetchAFHs = (afhIds) => {
+        let searchArray = new Set(afhIds);
+        API.get("api/askforhelp/all")
+            .then((res) => {
+                var userAFHs = [];
+                var afhs = [];
+                res.data.forEach((afh) => {
+                    if (searchArray.has(afh.id)) {
+                        userAFHs.push(afh);
+                    } else {
+                        afhs.push(afh);
+                    }
+                });
+
+                userAFHs = _.sortBy(userAFHs, ["date"]);
+                afhs = _.sortBy(afhs, ["date"]);
+                this.setState({
+                    userAFHs: userAFHs,
+                    afhs: afhs,
+                });
+            })
+            .catch((err) => alert("Could not fetch afhs: " + err));
+    };
+
+    onAFHChange = (e) => {
+        this.setState({
+            afhId: e.target.value,
+        });
+    };
+
+    onClickSchedule = () => {
+        const userAFH = {
+            userId: parseInt(this.state.id),
+            afhId: parseInt(this.state.afhId),
+        };
+
+        API.post("api/userafhs/create", userAFH)
+            .then(() => {
+                this.fetchUser();
+            })
+            .catch((err) => alert("Could not schedule AFH: " + err));
+    };
+
+    render = () => {
+        const user = this.state.user;
+
+        var fullName = user.firstName;
+        if (user.middleName) {
+            fullName += " " + user.middleName + " " + user.lastName;
+        } else {
+            fullName += " " + user.lastName;
+        }
```

Comment:
Maybe we should use a utility function for this.
Create a utility function (given a user object), construct the full name.
And have this utility function in the `utils` folder which can be easily imported by any other file.

Reply:
We have a similar logic above